src/agent/state/exporter.rs

```diff
@@ -534,46 +551,55 @@ where
 {
     let mut instructions = Vec::new();
 
-    // Refresh the data in the batch
-    let local_store_contents = LocalStore::get_all_price_infos(&*state).await;
-    let refreshed_batch = batch.iter().map(|(identifier, _)| {
-        (
-            identifier,
-            local_store_contents
-                .get(identifier)
-                .ok_or_else(|| anyhow!("price identifier not found in local store"))
-                .with_context(|| identifier.to_string()),
-        )
-    });
-    let price_accounts = refreshed_batch
-        .clone()
-        .map(|(identifier, _)| Pubkey::from(identifier.to_bytes()))
+    let price_accounts = batch
+        .iter()
+        .map(|update| Pubkey::from(update.feed_id.to_bytes()))
         .collect::<Vec<_>>();
 
-    for (identifier, price_info_result) in refreshed_batch {
-        let price_info = price_info_result?;
-        let now = Utc::now().naive_utc();
-
-        let stale_price = now > price_info.timestamp + staleness_threshold;
+    let now = Utc::now().naive_utc();
+    let mut updates = Vec::new();
+    // Refresh the data in the batch
+    let local_store_contents = LocalStore::get_all_price_infos(&*state).await;
+    for update in batch {
+        let mut update = update.clone();
+        update.info = local_store_contents
+            .get(&update.feed_id)
+            .ok_or_else(|| anyhow!("price identifier not found in local store"))
+            .with_context(|| update.feed_id.to_string())?
+            .clone();
+
+        let stale_price = now > update.info.timestamp + staleness_threshold;
         if stale_price {
             continue;
         }
+        updates.push(update);
+    }
 
+    if let Some(publish_program_key) = publish_program_key {
+        let (instruction, unsupported_updates) = create_instruction_with_publish_program(
+            publish_keypair.pubkey(),
+            publish_program_key,
+            updates,
+        )?;
+        updates = unsupported_updates;
```

Comment:
I think the fallback might fail because we also change the batch size, so maybe a simpler way can be just to crash? I think it highly depends on our upgrade process.